fuse_graphs/src/hash_graph.cpp

```diff
@@ -409,6 +409,22 @@ ceres::Solver::Summary HashGraph::optimize(const ceres::Solver::Options& options
   return summary;
 }
 
+void HashGraph::print(std::ostream& stream) const
+{
+  stream << "HashGraph\n"
+         << "  constraints:\n";
+  for (const auto& constraint : constraints_)
+  {
+    stream << "   - " << *constraint.second << "\n";
+  }
+  stream << "  variables:\n";
+  for (const auto& variable : variables_)
+  {
+    const auto is_on_hold = variables_on_hold_.find(variable.first) != variables_on_hold_.end();
+    stream << "   - " << (is_on_hold ? "[H] " : "") << *variable.second << "\n";
```

Comment:
For variables on hold I'm writting `[H] `, but I wonder if I could instead add another "field" like `on_hold: true` (or `false`). That way it'd be complaint with YAML, which seems to be the format you're trying to follow in all your outputs.

The downside though, is that it'd be harder to see if a variables it's on hold because the `on_hold` field would be at the bottom of the variable output. But for automated tools it'd be perfectly fine.

Reply:
I added the "on hold" feature because Ceres supported it. However, it did not do what I thought it would do. It acts as though you placed a prior with complete certainty on the "on hold" variables. This results in the variable values being computed as you might expect, but the uncertainty on the "on hold" variables is zero. As such, I'm not sure what the utility of holding a variable is. So, I wouldn't worry overly about the "on hold" being hard to see. I'm not sure how often it will ever get used.

Reply:
Thanks for the explanation. I've just changed this to the YAML compliant way: `on_hold: true/false`.